src/buffer/channel.rs

```diff
@@ -108,19 +113,151 @@ pub fn view<'a>(
         data::buffer::InputVisibility::Always => status.connected(),
     };
 
-    let text_input = show_text_input.then(|| {
-        column![
-            vertical_space(4),
-            input_view::view(
-                &state.input_view,
-                buffer,
-                users,
-                channels,
-                input_history,
-                is_focused
+    let topic_banner = config.buffer.topic_banner.enabled.then(|| {
```

Comment:
I feel like this part is quite complicated. We have nom inside view, and is doing quite a lot of calculations. We should tidy this up.

Reply:
Moved who/time line styling into its own function and removed nom for a much simpler split.  Had trouble finding documentation specifying the format for RPL_TOPICWHOTIME, but the simple split seems sufficient for our needs.